jboss-eap-config-jgroups/added/launch/jgroups.sh

```diff
@@ -0,0 +1,143 @@
+  configure
+}
+
+create_jgroups_elytron_encrypt_sym() {
+    declare jg_encrypt_keystore="$1" jg_encrypt_name="$2" jg_encrypt_password="$3" jg_encrypt_entire_message="$4"
+    local encrypt="<encrypt-protocol type=\"SYM_ENCRYPT\" key-store=\"${jg_encrypt_keystore}\" key-alias=\"${jg_encrypt_name}\">\
+                       <key-credential-reference clear-text=\"${jg_encrypt_password}\"/>\
+                       <property name=\"encrypt_entire_message\">${jg_encrypt_entire_message:-true}</property>\
+                   </encrypt-protocol>"
+    echo ${encrypt}
+}
+
+create_jgroups_encrypt_asym() {
+    # Asymmetric encryption using public/private encryption to fetch the shared secret key
+    # from the docs: "The ASYM_ENCRYPT protocol should be configured immediately before the pbcast.NAKACK2"
+    # this also *requires* AUTH to be enabled.
+    # TODO: make these properties configurable, this is currently just falling back on defaults.
+    declare encrypt_entire_message="$1" sym_keylength="$2" sym_algorithm="$3" asym_keylength="$4" asym_algorithm="$5" change_key_on_leave="$6"
+    local jgroups_encrypt="\
+                    <protocol type=\"ASYM_ENCRYPT\">\n\
+                        <property name=\"encrypt_entire_message\">${encrypt_entire_message:-true}</property>\n\
+                        <property name=\"sym_keylength\">${sym_keylength:-128}</property>\n\
+                        <property name=\"sym_algorithm\">${sym_algorithm:-AES/ECB/PKCS5Padding}</property>\n\
+                        <property name=\"asym_keylength\">${asym_keylength:-512}</property>\n\
+                        <property name=\"asym_algorithm\">${asym_algorithm:-RSA}</property>\n\
+                        <property name=\"change_key_on_leave\">${change_key_on_leave:-true}</property>\n\
+                    </protocol>"
+    echo ${jgroups_encrypt}
+}
+
+create_jgroups_elytron_legacy() {
+    declare jg_encrypt_keystore_dir="$1" jg_encrypt_keystore="$2" jg_encrypt_password="$3" jg_encrypt_name="$4"
+    # compatability with old marker, only used if new marker is not present
+    local legacy_encrypt="\
+        <protocol type=\"SYM_ENCRYPT\">\
+          <property name=\"provider\">SunJCE</property>\
+          <property name=\"sym_algorithm\">AES</property>\
+          <property name=\"encrypt_entire_message\">true</property>\
+          <property name=\"keystore_name\">${jg_encrypt_keystore_dir}/${jg_encrypt_keystore}</property>\
+          <property name=\"store_password\">${jg_encrypt_password}</property>\
+          <property name=\"alias\">${jg_encrypt_name}</property>\
+        </protocol>"
+    echo ${legacy_encrypt}
+}
+
+validate_keystore() {
+    declare jg_encrypt_secret="$1" jg_encrypt_name="$2" jg_encrypt_password="$3" jg_encrypt_keystore_dir="$4" jg_encrypt_keystore="$5"
+    if [ -n "${jg_encrypt_secret}"   -a \
+      -n "${jg_encrypt_name}"         -a \
+      -n "${jg_encrypt_password}"     -a \
+      -n "${jg_encrypt_keystore_dir}" -a \
```

Comment:
The keystore_dir is not required if "$(has_elytron_tls)" = "true". It's required for legacy.

Reply:
For now I've addressed this by adding a validate_keystore_legacy variant of this function, we can remove it once "legacy" support is no longer needed.